quesma/table_resolver/rules.go

```diff
@@ -9,34 +9,81 @@ import (
 	"quesma/end_user_errors"
 	"quesma/quesma/config"
 	"quesma/util"
-	"slices"
+	"reflect"
+	"strings"
 )
 
 // TODO these rules may be incorrect and incomplete
 // They will be fixed int the next iteration.
 
-func patternIsNotAllowed(input parsedPattern) *Decision {
-	if !input.isPattern {
-		return nil
-	}
-	return &Decision{
-		Reason: "Pattern is not allowed.",
-		Err:    fmt.Errorf("pattern is not allowed"),
+func (r *tableRegistryImpl) legacyPatternSplitter(pipeline string) func(pattern string) (parsedPattern, *Decision) {
+	return func(pattern string) (parsedPattern, *Decision) {
+		patterns := strings.Split(pattern, ",")
+
+		// Given a (potentially wildcard) pattern, find all non-wildcard index names that match the pattern
+		var matchingSingleNames []string
+		for _, pattern := range patterns {
+			if !elasticsearch.IsIndexPattern(pattern) || elasticsearch.IsInternalIndex(pattern) {
+				matchingSingleNames = append(matchingSingleNames, pattern)
+				continue
```

Comment:
Why `continue`?

Reply:
If `pattern` is not a pattern, but a single index, then there's no need to split it into single indexes (this is what code below does).

If `pattern` is an ElasticSearch internal index then we don't interfere with that (try to split it) - the `resolveInternalElasticName` will immediately decide to route it to Elastic. This is a bit of a workaround...

Reply:
ok, thx, now it's clear, however maybe short comment would be useful to avoid such questions

Reply:
Fixed in last commit - added comment.